### DISCUSSION TITLE
UARTinit() not called

### DISCUSSION THREAD

**Discussion by mstegen** [eModbus/eModbus#141] [Q&A]
Hi again, 

I'm trying arduino core 2.0.2 on my project, and noticed something strange.

In my project the ModbusServerRTU is listening for modbus data from other devices on the bus,  and processing data without sending any data back.
I noticed that the request and response interval time was not working correctly. Both messages were processed as one block of data, and the CRC check failed.
This was not a problem on arduino core 1.0.6. 

After some tests, i noticed that somehow `UARTinit()` is not called. At least i can not see the debug output "UARTinit: Serial1 FIFO threshold set to 1 (was 112)" this is for both 1.0.6 and 2.0.2 versions of arduino core.

So i added the call to `UARTinit()` in the `ModbusServerRTU::start` code. 
Now it's working on core2.0.2

**Comment by Miq1** [comment]
I was already expecting something in that area... The core maintainers have completely rewritten the UART handling in core 2.0+, using now more of the ESP-IDF implementation instead of their own. While I was hoping for some improvements in buffer handling, I always feared something we are using would be broken. And there it is...

I will have to closely look inside the new code.

**Comment by Miq1** [comment]
I opened [an issue](https://github.com/eModbus/eModbus/issues/142) for it.

**Reply by mstegen** [reply]
Thanks!
